src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Text/RuneTests.cs

```diff
@@ -57,7 +57,8 @@ public static void Casing_Invariant(int original, int upper, int lower)
             Assert.Equal(new Rune(lower), Rune.ToLowerInvariant(rune));
         }
 
-        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsIcuGlobalization), nameof(PlatformDetection.IsNotHybridGlobalizationOnBrowser))]
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsIcuGlobalizationAndNotHybridOnBrowser))]
+        // HybridGlobalization on Browser uses Invariant HashCode and SortKey, so its effect does not match this of ICU
```

Comment:
could you please elaborate ? Should this throw PNSE if it behaves wrongly ?

Reply:
It used to throw PNSE, but this behavior prevented a lot of APIs to be used, even far connected with Globalization, e.g. `System.Collections.Hashtable.Add()`. We decided to allow the APIs to work but with a restriction that they would share the HashCode and SortKey functions with Invariant mode. Because this test is not expected to work with Invariant mode (`PlatformDetection.IsIcuGlobalization` means that it should work only in ICU, not with Invariant), then it's a natural consequence that now for Hybrid it won't be expected to work either - it uses Invariant logic underneath.

Reply:
Should I maybe change the comment to be more descriptive? Like, "HybridGlobalization on Browser uses Invariant HashCode and SortKey, so its effect does not match this of ICU. This test is not run in InvariantGlobalizatio - IsIcuGlobalization=false then."

Reply:
I'm OK with the test change now. I'm also ok about `HashCode` because I understand that those could have collisions.

Reply:
I don't understand if different algo for SortKey would lead to different ordering in the results of the actual data. If so, that could be "wrong" and unexpected. I'm not sure what needs to be done about it.

Reply:
If we are now sorting invariant style in all places, that would be acceptable hybrid trade off. We just need to document it properly.

Reply:
Do you suggest adding info to docs/design/features/globalization-hybrid-mode.md? Or is it sufficient? There is a list of known APIs that might be affected + info that Invariant SortKey is used for them.

Reply:
Hmm, about `HashCode`, it would be OK to have more collisions, but this is making less collisions. Things which do string equal don't do hash equals. That breaks the the contract, I think.